examples/scripts/ct_projector_comparison_3d.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This file is part of the SCICO package. Details of the copyright
+# and user license can be found in the 'LICENSE.txt' file distributed
+# with the package.
+
+
+r"""
+X-ray Transform Comparison in 3D
+================================
+
+This example shows how to define a SCICO native 3D X-ray transform using
+ASTRA toolbox conventions and vice versa.
+"""
+
+import numpy as np
+
+import jax
+import jax.numpy as jnp
+
+import scico.linop.xray
+import scico.linop.xray.astra as astra
+from scico import plot
+from scico.examples import create_block_phantom
+from scico.linop import Parallel3dProjector, XRayTransform
+from scico.util import Timer
+from scipy.spatial.transform import Rotation
+
+"""
+Create a ground truth image and set detector dimensions.
+"""
+N = 64
+# use rectangular volume to check whether it is handled correctly
+in_shape = (N + 1, N + 2, N + 3)
+x = create_block_phantom(in_shape)
+x = jnp.array(x)
+
+# use rectangular detector to check whether it is handled correctly
+out_shape = (N, N + 1)
+
+
+"""
+Set up SCICO projection.
+"""
+num_angles = 3
+
+# make projection matrix: form a rotation matrix and chop off the last row
+rot_X = 90.0 - 16.0
+rot_Y = np.random.rand(num_angles) * 180
+P = jnp.stack([Rotation.from_euler("XY", [rot_X, y], degrees=True).as_matrix() for y in rot_Y])
+P = P[:, :2, :]
+
+# add translation
+x0 = jnp.array(in_shape) / 2
+t = -jnp.tensordot(P, x0, axes=[2, 0]) + jnp.array(out_shape) / 2
+P = jnp.concatenate((P, t[..., np.newaxis]), axis=2)
+
+
+"""
+Specify geometry using SCICO conventions and project.
+"""
+num_repeats = 3
+
+timer_scico = Timer()
+timer_scico.start("init")
+H_scico = XRayTransform(Parallel3dProjector(in_shape, P, out_shape))
+timer_scico.stop("init")
```

Comment:
This would be a bit more compact as
```python
timer_scico = Timer()
with ContextTimer(timer_scico, "init"):
    H_scico = XRayTransform(Parallel3dProjector(in_shape, P, out_shape))
```

Reply:
Done.